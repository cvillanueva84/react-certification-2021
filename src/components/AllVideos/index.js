export { default } from './AllVideos.component';