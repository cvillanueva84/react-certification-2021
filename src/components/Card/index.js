export {default} from './Card.component.jsx'; 