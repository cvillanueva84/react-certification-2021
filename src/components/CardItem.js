import React from 'react';
import { useContext } from 'react';
import { GlobalContext } from './Context';
import {CardItemUi}  from './ui/CardItemUi'

export const CardItem = ({ title, description, url, videoId }) => {

  const { myStateReducer, dispatch } = useContext(GlobalContext);

  const handleClick = () => {
      dispatch({
        type: 'actionAddVideoData',
        payload: {
          ...myStateReducer,
          videoData: {
            title: title,
            description: description,
            url: url,
            videoId: videoId
          }
        }
      });
  }

  return (
    <CardItemUi 
      title={title} 
      description={description} 
      url={url} 
      handleClick={handleClick}
    />
  );

};

