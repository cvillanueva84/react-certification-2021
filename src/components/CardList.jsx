import React from 'react';
import Grid from '@material-ui/core/Grid';
import VideoCard from './VideoCard';
import videos from '../youtube-videos-mock';

export default function CardList() {
  return (
    <Grid
      container
      justifyContent="center"
      alignItems="center"
      style={{ padding: '30px' }}
    >
      {videos.items.map((video) => (
        <VideoCard videoInfo={video.snippet} key={video.etag} />
      ))}
    </Grid>
  );
}
