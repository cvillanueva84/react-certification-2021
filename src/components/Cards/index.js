export { default } from './Cards.component';
