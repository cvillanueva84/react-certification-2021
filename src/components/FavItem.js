import React, { useContext } from 'react';
import '../style/FavoriteVideosViewStyled.js';
import { GlobalContext } from './Context';
import { FavItemUi } from './ui/FavItemUi.js';


export const FavItem = ({ title, description, url, videoId }) => {
  const { myStateReducer, dispatch } = useContext(GlobalContext);

  const handleClickPlay = () => {
    dispatch({
      type: 'actionAddVideoDataFav',
      payload: {
        ...myStateReducer,
        videoDataFav: {
          title: title,
          description: description,
          url: url,
          videoId: videoId,
        },
      },
    });
  };

  const handleClickDelete = () => {
    dispatch({
      type: 'actionDelete',
      payload: videoId,
    });
  };

  return (
    <FavItemUi 
      title={title} 
      url={url} 
      description={description} 
      handleClickPlay={handleClickPlay} 
      handleClickDelete={handleClickDelete}
    />
  );
};

