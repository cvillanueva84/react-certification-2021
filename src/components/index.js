export { default as Navbar } from './Navbar/Navbar';
export { default as SearchBar } from './SearchBar/SearchBar';
export { default as ThemeSwitch } from './ThemeSwitch/ThemeSwitch';
export { default as Results } from './Results/Results';
