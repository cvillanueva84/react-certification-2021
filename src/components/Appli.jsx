import React from 'react';
import CardList from './CardList';
import NavBar from './NavBar';

export default function Appli() {
  return (
    <div>
      <NavBar />
      <CardList />
    </div>
  );
}
