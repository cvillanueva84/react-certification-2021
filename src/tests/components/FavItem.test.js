import React from 'react';
import '@testing-library/jest-dom';
import { mount, render } from 'enzyme';
import { GlobalContext } from '../../components/Context';
import { MemoryRouter, Router } from 'react-router';
import { FavItem } from '../../components/FavItem';

describe('Test FavItem', () => {
    const contextValue = {
        dispatch: jest.fn(),
        myStateReducer: {
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        }
    }
    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    }

    const wrapper = mount(
        <GlobalContext.Provider value={contextValue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <FavItem />
                </Router>
            </MemoryRouter>
        </GlobalContext.Provider>
    );

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render success', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.exists());
    });

    it('should have elements', () => {
        expect(wrapper.find('img').length).toBe(1);
        expect(wrapper.find('h2').length).toBe(1);
        expect(wrapper.find('p').length).toBe(1);
        expect(wrapper.find('button').length).toBe(2);
    });

    it('should dispatch videodetailsfavs', () => {
        wrapper.find('a').simulate('click')
        expect(contextValue.dispatch).toHaveBeenCalled();
    });

    it('should dispatch actionDelete', () => {
        wrapper.find('#delete').simulate('click')
        expect(contextValue.dispatch).toHaveBeenCalled();
    });

});