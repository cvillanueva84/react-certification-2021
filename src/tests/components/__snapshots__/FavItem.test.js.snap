// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FavItem should render success 1`] = `ReactWrapper {}`;
